src/analysis/special_functions/exponential.lean

```diff
@@ -211,3 +218,183 @@ end
 
 lemma real.exp_eq_exp_ℝ : real.exp = exp ℝ :=
 by { ext x, exact_mod_cast congr_fun complex.exp_eq_exp_ℂ x }
+
+/-! ### Derivative of $\exp (ux)$ by $u$
+
+Note that since for `x : 𝔸` we have `normed_ring 𝔸` not `normed_comm_ring 𝔸`, we cannot deduce
+these results from `has_fderiv_at_exp_of_mem_ball`.
+
+We could deduce them from the more general non-commutive case,
+$$\frac{d}{dt}e^{x(t)} = \int_0^1 e^{sx(t)} \left(\frac{d}{dt}e^{x(t)}\right) e^{(1-s)x(t)} ds$$
+but this is harder to prove, and typically is shown by going via these results first.
```

Comment:
```suggestion
Note that since for `x : 𝔸` we have `normed_ring 𝔸` not `normed_comm_ring 𝔸`, we cannot deduce
these results from `has_fderiv_at_exp_of_mem_ball` applied to the algebra `𝔸`.

One possible solution for that would be to apply `has_fderiv_at_exp_of_mem_ball` to the
commutative algebra `algebra.elemental_algebra 𝕊 x`. Unfortunately we don't have all the required
API, so we leave that to a future refactor (see leanprover-community/mathlib#19062 for discussion).

We could also go the other way around and deduce `has_fderiv_at_exp_of_mem_ball` from
`has_fderiv_at_exp_smul_const_of_mem_ball` applied to `𝕊 := 𝔸`, `x := (1 : 𝔸)`, and `t := x`.
However, doing so would make the aformentioned `elemental_algebra` refactor harder, so for now we
just prove these two lemmas independently.

A last strategy would be to deduce everything from the more general non-commutive case,
$$\frac{d}{dt}e^{x(t)} = \int_0^1 e^{sx(t)} \left(\frac{d}{dt}e^{x(t)}\right) e^{(1-s)x(t)} ds$$
but this is harder to prove, and typically is shown by going via these results first.
```

Okay I may have been a bit carried away while writing this, but at least everything is written explicitly :sweat_smile: Does that work for you?